keps/sig-network/4858-ip-cidr-validation/README.md

```diff
@@ -0,0 +1,838 @@
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+It is automatically in use in any cluster where it is enabled.
+
+###### How can someone using this feature know that it is working for their instance?
+
+"Things didn't break"
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+We could add a metric to kube-apiserver for when a Create/Update gets
+an IP/CIDR-related warning. That might help admins decide whether it
+was safe to upgrade, though it wouldn't especially help them figure
+out _why_. A more useful approach might be to make sure that the
+warnings get logged, with information about specific objects, which
+would be more useful in helping to track down the offending
+users/controllers.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+The KEP will add new code to the Create/Update path (to check objects
+and generate the warnings) for certain types (including Pod and
+Service), but it is no heavier than any other increased validation.
+
+The intent is to try to minimize additional validation on Endpoints
+and EndpointSlice, since those objects can be particularly large (and
+generally don't need extra validation).
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+It shouldn't.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A; it is part of the API server.
+
+###### What are other known failure modes?
+
+None
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A.
+
+## Implementation History
+
+- 2024-01-01: Proposed as a PR ([kubernetes #122550])
+- 2024-10-02: Proposed as a KEP
+
+## Drawbacks
+
+The drawback of implementing this is that it may require some users to
```

Comment:
This section is to explain what can happen if this is NOT implemented.

Reply:
I feel like you are saying the same thing?  I don't think there are any drawbacks to implementing, only some risks.  If we make it past the risks, there are no lingering detrimental effects, right?

Reply:
I'm explaining the Drawbacks of implementing this KEP. Maciej seemed to be saying that I should be explaining the Drawbacks of NOT implementing the KEP.